nestkernel/structural_plasticity_node.h

```diff
@@ -0,0 +1,184 @@
+/*
+ *  structural_plasticity_node.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef STRUCTURAL_PLASTICITY_NODE_H
+#define STRUCTURAL_PLASTICITY_NODE_H
+
+// C++ includes:
+#include <algorithm>
+#include <deque>
+
+// Includes from nestkernel:
+#include "histentry.h"
+#include "nest_time.h"
+#include "nest_types.h"
+#include "node.h"
+#include "synaptic_element.h"
+
+// Includes from sli:
+#include "dictdatum.h"
+
+#define DEBUG_ARCHIVER 1
+
+namespace nest
+{
+
+/**
+ * \class Structural_Plasticity_Node
+ * a node which archives spike history for the purposes of
```

Comment:
Single line summary starting with capital letter. Details can follow in a separate paragraph.